src/message_handler/src/messages.rs

```diff
@@ -0,0 +1,98 @@
+use futures::{channel::mpsc::channel, channel::mpsc::Receiver, channel::mpsc::Sender};
+use libp2p::floodsub::{protocol, Topic, TopicBuilder, TopicHash};
+use sdk::constants;
+use std::sync::{Arc, Mutex};
+use utils::serializer::{deserialize, Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum MessageTypes {
+    NodeMsg(Vec<u8>),
+    ConsensusMsg(Vec<u8>),
+}
+
+//TODO : Try using macro to implement this for all variations
+impl From<MessageTypes> for Vec<Topic> {
+    fn from(msg: MessageTypes) -> Vec<Topic> {
+        let mut ret: Vec<Topic> = Vec::new();
+        match msg {
+            MessageTypes::NodeMsg(_data) => {
+                ret.push(TopicBuilder::new(constants::NODE).build());
+            }
+            MessageTypes::ConsensusMsg(_data) => {
+                ret.push(TopicBuilder::new(constants::CONSENSUS).build());
+            }
+        }
+        ret
+    }
+}
+
+///Process FloodSubMessages
+///
+pub trait MsgProcess {
+    fn process(&self, topics: &Vec<TopicHash>, data: &Vec<u8>);
+}
+
+impl MsgProcess for protocol::FloodsubMessage {
+    fn process(&self, topics: &Vec<TopicHash>, data: &Vec<u8>) {
+        if topics[0] == TopicBuilder::new(constants::NODE).build().hash().clone() {
+            debug!("NodeMessageTypes data received");
+            if let Ok(deserialize_msg) = deserialize::<Vec<u8>>(data) {
+                let result = MSG_DISPATCHER
+                    .node_msg_dispatcher
+                    .clone()
+                    .try_send(Some(deserialize_msg));
+                if result.is_err() {
+                    result.unwrap_err().into_send_error();
+                }
+            }
+        } else if topics[0]
+            == TopicBuilder::new(constants::CONSENSUS)
+                .build()
+                .hash()
+                .clone()
+        {
+            debug!("ConsensusMessageTypes data received");
+            if let Ok(deserialize_msg) = deserialize::<Vec<u8>>(data) {
+                let result = MSG_DISPATCHER
+                    .consensus_msg_dispatcher
+                    .clone()
+                    .try_send(Some(deserialize_msg));
+                if result.is_err() {
+                    result.unwrap_err().into_send_error();
+                }
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct MessageDispatcher {
+    pub node_msg_dispatcher: Sender<Option<Vec<u8>>>,
```

Comment:
Sender/Receiver should be of strict data type instead of generic byte array.
If byte array we may not need sendear/receiver per message type
Need to have some mechanism to auto-add sender/receiver when new message type is added int the system with no code change here.